src/stdlib/SDL_string.c

```diff
@@ -28,6 +28,205 @@
 #include <psp2/kernel/clib.h>
 #endif
 
+#include "SDL_casefolding.h"
+
+// this is the Unicode REPLACEMENT CHARACTER, used for invalid codepoint values.
+#define INVALID_UNICODE_CODEPOINT 0xFFFD
+
+#if defined(__SIZEOF_WCHAR_T__)
+#define SDL_SIZEOF_WCHAR_T __SIZEOF_WCHAR_T__
+#elif defined(SDL_PLATFORM_WINDOWS)
+#define SDL_SIZEOF_WCHAR_T 2
+#else  // assume everything else is UTF-32 (add more tests if compiler-assert fails below!)
+#define SDL_SIZEOF_WCHAR_T 4
+#endif
+SDL_COMPILE_TIME_ASSERT(sizeof_wchar_t, sizeof(wchar_t) == SDL_SIZEOF_WCHAR_T);
+
+
+// this expects `from` and `to` to be UTF-32 encoding!
+static int SDL_UnicodeCaseFold(const Uint32 from, Uint32 *to)
+{
+    // !!! FIXME: since the hashtable is static, maybe we should binary
+    // !!! FIXME: search it instead of walking the whole bucket.
+
+    if (from < 128) {   // low-ASCII, easy!
+        if ((from >= 'A') && (from <= 'Z')) {
+            *to = from - ('A' - 'a');
```

Comment:
entirely frivolous, but this is easier for me to read:
```c
*to = 'a' + (from - 'A');
```

Reply:
Changed in c69eac4da8811ecb01b313a6c620feb33ea631bf